tasks/tools.py

```diff
@@ -49,12 +53,113 @@ def remove_from_do(id):
 
     session.execute("COMMIT")
 
+
+def extract_nulls_from_log(infile, outfile, relevant_fields):
+    STR_NULL = 'contain only NULL values: '
+    STR_ERROR = '[ERROR]'
+    STR_INFO = '[INFO]'
+    STR_EXCEPTIONS = 'exceptions'
+
+    relevant_fields = relevant_fields.split(',')
+    cols = {}
+
+    try:
+        os.remove(outfile)
+    except OSError:
+        pass
+
+    with open(outfile, 'a') as fout:
+        with open(infile) as fin:
+            current_params = {}
+            for line in fin:
+                if STR_ERROR in line:
+                    current_params = {}
+                    try:
+                        vals = {item.split("=")[0].strip(): item.split("=")[1].strip()
+                                for item in line[line.rfind('(') + 1:line.rfind(')')].split(',')}
+                        for key, val in vals.items():
+                            if key in relevant_fields:
+                                current_params[key] = val
+                    except Exception as e:
+                        pass
+
+                elif STR_NULL in line and STR_INFO not in line:
+                    columns = [x.strip() for x in line[line.rfind(STR_NULL) + len(STR_NULL):].split(',')]
+
+                    for column in columns:
+                        if column in cols:
+                            if current_params not in cols[column][STR_EXCEPTIONS]:
+                                cols[column][STR_EXCEPTIONS].extend([current_params])
+                        else:
+                            cols[column] = {}
+                            cols[column][STR_EXCEPTIONS] = [current_params]
+
+        fout.write(json.dumps(cols, indent=4))
+
+    print(' >>>---> Done! Written file {}'.format(outfile))
+
+
+def merge_nulls_files(infile1, infile2, outfile):
+    STR_EXCEPTIONS = 'exceptions'
+
+    def _merge_item(item, itemlist):
+        for iteml in itemlist:
+            if item == iteml:
+                break
+        else:
+            return item
+
+        return None
+
+    try:
+        os.remove(outfile)
+    except OSError:
+        pass
+
+    with open(outfile, 'a') as fout:
+        with open(infile1) as fin1:
+            json1 = json.load(fin1)
+            with open(infile2) as fin2:
+                json2 = json.load(fin2)
+
+                # The ones from infile1 that are also in infile2
+                for column1 in json1:
```

Comment:
An alternative to these two loop is to iterate over all columns in both jsons and merge as needed:
```
for column in set(json1.keys() + json2.keys()):
    col1 = json1.get(column, json1.get(column.lower(), []))
    col2 = json2.get(column, json2.get(column.lower(), []))
    json1[column] = col1 + col2
```

This will generate in json1 a dictionary with all keys from json1 and json2. For keys that are in both, it will add the values without repetition.